Add more informative error message for invalid `unroll` parameter in `lax.scan`

dfm opened the pull request:
As reported in #20481, setting `unroll=0` in `lax.scan` resulted in an uninformative `ZeroDivisionError`. This PR adds a check which raises a `ValueError` for `unroll<=0`.

One interesting (to me!) technical point is that a more informative error is raised when `jax_enable_checks` is on ([ref](https://github.com/google/jax/blob/4ccac4c6ce93bfd69a01699c11a30afa35e2ef11/jax/_src/lax/control_flow/loops.py#L1165)), but since this is off by default, I think it is sensible to have a better error message. That being said, I'm not sure if the way I'm turning `jax_enable_checks` off in the test is appropriate.

jakevdp commented on tests/lax_control_flow_test.py:
```diff
@@ -1951,6 +1951,21 @@ def testScanBodyCarryTypeMismatchErrors(self):
+  def testScanInvalidUnrollRaises(self):
+    # 'jax_enable_checks' must be set to `False` to trigger this error because
+    # otherwise the invalid value for unroll is caught by `_scan_typecheck`.
+    prev = jax.config.jax_enable_checks
```
You can use `with jax.enable_checks(False)` to do this automatically; or alternatively decorate the test method with `@jax.enable_checks(False)`

dfm replied:
Thanks! I couldn't find that in an initial search. Updated!

jakevdp reviewed (approved):
Looks good! Just a quick comment below about simplifying the `enable_checks` setting.

Also, once you make that update please squash the changes into a single commit (we do that to keep the commit history clean, because our tooling can't currently squash automatically).

dfm committed:
Add informative error for invalid unroll in scan

As reported in #20481, setting `unroll=0` in `lax.scan` resulted in an
uninformative `ZeroDivisionError`. This PR adds a check which raises a
`ValueError` for `unroll<=0`.
